docs/java-rest/high-level/apis/bulk.asciidoc

```diff
@@ -115,3 +117,95 @@ include-tagged::{doc-tests}/CRUDDocumentationIT.java[bulk-errors]
 --------------------------------------------------
 <1> Indicate if a given operation failed
 <2> Retrieve the failure of the failed operation
+
+[[java-rest-high-document-bulk-processor]]
+==== Bulk Processor
+
+The `BulkProcessor` simplifies the usage of the Bulk API by providing
+a utility class that allows index/update/delete operations to be
+transparently executed as they are added to the processor.
+
+In order to execute the requests, the `BulkProcessor` requires 3 things:
+
+`RestHighLevelClient`:: This client is used to execute the `BulkRequest`s
+and to retrieve the `BulkResponse`s
```

Comment:
do we want to mention that the bulkAsync method will be called internally (as we provide it when creating the processor)